cmd/handler-api.go

```diff
@@ -141,7 +141,11 @@ func maxClients(f http.HandlerFunc) http.HandlerFunc {
 
 		select {
 		case pool <- struct{}{}:
-			defer func() { <-pool }()
+			defer func() {
+				<-pool
+				globalHTTPStats.addRequestsInQueue(-1)
+			}()
+			globalHTTPStats.addRequestsInQueue(1)
```

Comment:
This is incorrect. You are counting running requests, not queued requests.

Queued requests are blocked on `pool <- struct{}{}` so they will only arrive here when executing.

Reply:
Could just be semantics, but `s3RequestsInQueue` should be `s3RequestsRunning` or similar.

Reply:
You are right, it is fixed now, but still keeping queue semantic, please review again :)